osf/models/node.py

```diff
@@ -2787,30 +2790,33 @@ def remove_node(self, auth, date=None):
         :param datetime date: `datetime.datetime` or `None`
         """
         # TODO: rename "date" param - it's shadowing a global
-        if not self.can_edit(auth):
+        hierarchy = Node.objects.get_children(self, active=True, include_root=True)
+        is_admin = Contributor.objects.filter(node=OuterRef('pk'), admin=True, user=auth.user)
+        if len(hierarchy.annotate(has_admin_perm=Exists(is_admin)).filter(has_admin_perm=False)):
```

Comment:
Calling len instead of count(), since we have to iterate over the queryset anyway.